File: Python/SQLAlchemy_ORM/bll/services/order_service.py
from dal.repositories.order_repository import OrderRepository
from dal.repositories.test_datum_repository import TestDatumRepository
from dal.data.database import SessionLocal
from dal.entities.order import Order

class OrderService:
    def __init__(self):
        self.db = SessionLocal()
        self.repo = OrderRepository(self.db)

    def get_by_id(self, id: int):
        return self.repo.get_by_id(id)

    def get_all(self):
        return self.repo.get_all()

    def create(self, entity: Order) -> Order:
        return self.repo.create(entity)
    
    def update(self, id: int, entity: Order) -> Order:
        return self.repo.update(id, entity)

    def delete(self, id: int):
        return self.repo.delete(id)

File: Node/Sequelize_ORM/src/bll/services/orderService.js
const OrderRepository = require('../../dal/repositories/orderRepository');

class OrderService {
  async getAll() {
    return await OrderRepository.getAll();
  }

  async getById(id) {
    return await OrderRepository.getById(id);
  }

  async create(entity) {
    return await OrderRepository.create(entity);
  }

  async update(id, entity) {
    await OrderRepository.update(id, entity);
  }

  async delete(id) {
    await OrderRepository.delete(id);
  }

  async allTables() {
    await OrderRepository.allTables();
  }
}


module.exports = new OrderService(); 